src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -193,7 +193,7 @@ private SyntaxBase ParseJTokenExpression(JTokenExpression expression)
             => expression.Value.Type switch
             {
                 JTokenType.String => SyntaxFactory.CreateStringLiteral(expression.Value.Value<string>()!),
-                JTokenType.Integer => new IntegerLiteralSyntax(SyntaxFactory.CreateToken(TokenType.Integer, expression.Value.ToString()), expression.Value.Value<long>()),
+                JTokenType.Integer => new IntegerLiteralSyntax(SyntaxFactory.CreateToken(TokenType.Integer, expression.Value.ToString()), expression.Value.Value<ulong>()),
```

Comment:
I think we'll need to separate out the -ve and +ve cases here, as otherwise we'll end up casting a negative value to a positive one, or causing an exception - similar to the logic you changed in `InsertResourceHandler.cs`

Reply:
BTW - I see we've already got a method `ParseIntegerJToken` in this class - I think we should be able to use it here instead of calling `new IntegerLiteralSyntax(...)`. and implement the -ve/+ve handling in `ParseIntegerJToken` - following your `InsertResourceHandler` logic:
![image](https://user-images.githubusercontent.com/38542602/152048763-37d96c99-6a03-4ec2-b8d0-02ef84aff7e1.png)